### PR TITLE
minor refactoring on replica-server and meta-server

### PR DESCRIPTION
**Pull request by neverchanje**
What this PR did:

Remove some of the useless functions in replica-server:

- `replication_app_base::reset_counters_after_learning`
- `replication_app_base::install_perf_counters`
- `mutation_cache::remove_mutation_by_decree`

Some functions are refactored:

- `replica::query_app_envs`
- `replica::update_app_envs`

Rename  `update_commit_statistics` to `update_commit_qps`, `_counter_replicas_total_commit_throught` to `_counter_replicas_commit_qps`.

### CODE REVIEW

File: src/dist/replication/lib/replica.h [hunk 1]
```diff
@@ -233,9 +237,10 @@ class replica : public serverlet<replica>, public ref_counter, public replica_ba
                                               partition_status::type news);
 
     // return false when update fails or replica is going to be closed
-    bool update_app_envs(const std::map<std::string, std::string> &envs);
+    void update_app_envs(const std::map<std::string, std::string> &envs);
```
Comment (qinzuoyan) [inline src/dist/replication/lib/replica.h]: 上面的注释去掉

File: src/dist/replication/lib/replica_stub.cpp [hunk 2]
```diff
@@ -2010,11 +2010,8 @@ void replica_stub::open_service()
         [this](const std::vector<std::string> &args) {
             return exec_command_on_replica(args, true, [](const replica_ptr &rep) {
                 std::map<std::string, std::string> kv_map;
-                if (rep->query_app_envs(kv_map)) {
-                    return dsn::utils::kv_map_to_string(kv_map, ',', '=');
-                } else {
-                    return std::string("call replica::query_app_envs() failed");
-                }
+                rep->query_app_envs(kv_map);
```
Comment (qinzuoyan) [inline src/dist/replication/lib/replica_stub.cpp]: 这里不区分查询失败的情况，合适吗？

Comment (neverchanje) [reply src/dist/replication/lib/replica_stub.cpp]: 一个环境变量不应该设置失败，如果要失败应该在 meta server 就失败。
同样现在也没有查询环境变量失败的场景，以后有就以后再处理吧。

File: src/dist/replication/lib/replica.h (hunk 1 above)
Comment (neverchanje) [reply src/dist/replication/lib/replica.h]: fixed

**Pull request merged**